### ISSUE TITLE
Camera compatibility with DFRobot low light camera

### ISSUE THREAD

**Issue by nmg196** [motioneye-project/motioneye#2438]
Does anyone know if MotionEye should be compatible with this camera:
https://wiki.dfrobot.com/Night%20Camera%20Module%20for%20Raspberry%20Pi%20SKU:%20SEN0338

The camera works with it's own software (veye) but in MotionEye, I don't see any cameras recognised under MMAL or V4L2 :(

Thanks!
Nick

**Comment by nmg196** [comment]
Ahh OK - I got it working by installing the V4L2 driver using these instructions: http://wiki.veye.cc/index.php/V4L2_mode_for_Raspberry_Pi

The frame rate doesn't seem that great though. Looks like it's about 5 FPS even when set to 25 in MotionEye.

**Comment by MichaIng** [comment]
Is the original camera stream as well at 5 FPS? You should see one value for the original stream and one served by motionEye. Of course it can be lower due to network limitations e.g.

**Comment by nmg196** [comment]
Hmm I'm not sure where I'd see those figures?  I doubt it's network limitations as the Pi is plugged into Gigabit ethernet.

**Comment by MichaIng** [comment]
The figures are only visible when the stream is there, of course, i.e. you need to set the resolution to a supported one. The supported resolutions for V4L2 cameras are btw obtained via:
```sh
v4l2-ctl -d /dev/video0 --list-formats-ext | grep -vi stepwise | grep -oE '[0-9]+x[0-9]+'
```
if the camera is provided via `/dev/video0`. If for some reason this doesn't output anything, a common resolution list is shown, which doesn't say whether any of them is really supported or not.

Another question, do you use the current stable Python 2 version of motionEye, or the new Python 3 compatible version (not released to PyPI yet), installable via [these instructions](https://github.com/motioneye-project/motioneye/tree/dev#installation)?

**Comment by nmg196** [comment]
Hi,
It seems my camera only supports 1920x1080

![image](https://user-images.githubusercontent.com/8134047/165915518-1218d0a8-feaf-4d9e-8868-b90faa4768c1.png)

I'm using the Python 3.7 version from those instructions. 

I'm also not getting any recordings for some reason with these settings:

![image](https://user-images.githubusercontent.com/8134047/165916387-6552b680-db3e-4ab9-babf-071cc812ac27.png)

Thanks

**Comment by MichaIng** [comment]
Can you please share the service logs:
```sh
journalctl -u motioneye
```

**Comment by nmg196** [comment]
Yeah here's a screenshot of today's log. The red part repeats for hours:

![image](https://user-images.githubusercontent.com/8134047/165971162-4cc16ac2-fe78-4bd8-844a-2a2bf589d051.png)

**Comment by MichaIng** [comment]
You need to select a different "Movie format" in the settings. AFAIK H.264/OMX one is only supported with the Raspberry Pi FFmpeg build. Would actually make sense to check for this once at motionEye start and show a the regular H.264 codec if it's not an RPi, respectively if `ffmpeg` doesn't list the OMX as supported format.

**Comment by nmg196** [comment]
Thanks - that fixed it :)

**Comment by nmg196** [comment]
The frame rate is still really slow for some reason. I will post a sample video so you can see what I mean. The stream is reported at being close to 29FPS, but in reality it's nothing close to that - maybe 4-5 frames per second.  I'm wondering if the exposure duration is actually 1/5th of a second or something and that's causing the 5FPS effect, but that wouldn't explain the 29FPS reported by motionEye.

**Comment by MichaIng** [comment]
There should be two frame rates shown in motionEye, one is the intended frame rate and one is what finally appears at the frontend. Are both reported at close to 29 FPS?

**Comment by nmg196** [comment]
Yes, both say 30ish: 
![image](https://user-images.githubusercontent.com/8134047/166465653-b2dc06c1-b735-4e9f-836f-0a3bf94b3a51.png)

**Comment by MichaIng** [comment]
Okay, then I'm not sure whether it's the camera itself providing a new picture only at 5 FPS while it is possible to grab (then repeating) pictures at 30 FPS. Clarification could bring watching the `motion` stream directly instead of the motionEye stream, and testing with another camera software, like mjpg-streamer, whether it similarly appears to be as well 5 FPS in real.
